PyInstaller/__main__.py

```diff
@@ -177,6 +184,7 @@ def run(pyi_args: list | None = None, pyi_config: dict | None = None):
         else:
             spec_file = run_makespec(**vars(args))
 
+        sys.argv = [spec_file, *spec_args]
```

Comment:
Should we be restoring `sys.argv` back to the original value once the spec execution finishes?

Reply:
Not sure. It would be more hygienic but I can't think of a scenario where it would matter. `sys.argv` should never be read again.